### ISSUE TITLE
Suggestion: Additional context menu commands for folders

### ISSUE THREAD

**Issue by carloscadux** [fredjeck/fav#20]
Hi fredjeck,

A suggestion regarding additional context menu commands for folders, once you have a folder as favorite (after using command 'Add folders to favorites...')

As illustrated in the image below, I'm referring to two commands:

(a) Open folder files in new window:
A new VSCode window is loaded and the files (according to Fav filtering) are opened
No ‘folder is opened’ (single-folder workspace)

(b) Open single-folder in new window:
A new VSCode window is loaded and the files (according to Fav filtering) are opened
No ‘folder is opened’ (single-folder workspace)

Regarding (a), it's a unique functionality among other Favorite extensions, since it is an extended command of your handy implementation of opening files within the folder given a certain filter.

Regarding (b), other Favorite extensions generally perform only this command to folders added as favorites. After having installed Fav, I've kept other favorite extension just to perform this command. So having this command in Fav means I can uninstall other extension and have only Fav for managing favorites :)

If you need further information, just let me know.

Cheers,
Carlos

![image](https://user-images.githubusercontent.com/7339389/156217605-38f3e36b-b3b3-4434-adb3-7752068cef3a.png)

**Comment by fredjeck** [comment]
@carloscadux I don't really understand the difference between both approaches.

While opening a whole folder in a new window is pretty obvious using the built in commands, it is a totally different story when it comes to open multiple files in a new window as the extension API does not give any control for that. I am presently search for clean ways of implementing this feature and use as much as possible the extension API and avoid having to manually create new VSCode processes

**Comment by carloscadux** [comment]
Hey @fredjeck 

**Regarding (a)**, **Open folder files in new window**, it is the sibling functionality, aimed to folders, which complements this other suggestion:
[Additional context menu command for files (Open in new window) · Issue #21](https://github.com/fredjeck/fav/issues/21). 

I was considering two different elements, the favorites added through `...active files to favorites` and `...folders to favorites`, because of that I created separated suggestion.

Anyway, if the 'Open in new window' command to be implemented, which you mentioned on issue [Issue #21](https://github.com/fredjeck/fav/issues/21), will cover all favorite elements, that's nice. In that case, please ignore request (a) in this thread.

**Regarding (b)**, **Open single-folder in new window**, although "opening a whole folder in a new window is pretty obvious using the built in commands", it is time consuming clicking `File -> Open...`, looking for the folder, and finally having it opened as a single-folder workspace.

So considering all my favorite folders are already set in 'Fav', this command would speed up workflow.

In short (a) is totally aligned with your functionality `...folders to favorites`, which would open in a new window the group of files controlled by a filtered folder; (b) adds single-folder workspaces management straight from the favorite lists, as you can see in this other favorite extension demonstration I recorded: https://bit.ly/3pyUtD1

**Comment by fredjeck** [comment]
Taking the next version for a test drive.
I am not satisfied with the "Open files in a new window" but this is the closest I can get to the  described feature due to the lack of support of VSCode extension API for opening files in a new window instance (currently it only supports a single file or a single folder). The delay you can see comes from the fact that the extension needs to be activated to kinda "restore" the files from the previously selected folder in the other instance of the extension.
![Sample](https://user-images.githubusercontent.com/15845066/156939507-e71d0a7f-bf68-4c1b-b1fc-cf4ede8b0e71.gif)

**Comment by fredjeck** [comment]
Proper resolution will only be possible when https://github.com/microsoft/vscode/issues/122071 will be fixed

**Comment by fredjeck** [comment]
I created a pre-release if you want to give it a spin : https://github.com/fredjeck/fav/releases/tag/1.7.0-b01

**Comment by carloscadux** [comment]
Thanks for implementing the functionalities and letting me know about the pre-release 👍
I just gave it a try. 

a) I created a favorite folder `*.*` with several files without extension. 
When clicking on the folder, the files aren't opened within the current session. Nothing happened because of the extension lacking, it seems.
When right-clicking and choosing `'open files in new window'`, no file is opened. There is an unexpected folder selection window popping up though. 
When right-clicking and choosing `'open in new window'`, the command worked properly, loading a single-folder workspace.

b) Because of the limitation you mentioned of the APPI opening just a single file when the command `'open files in new window'` is triggered from a filtered folder `*.*`, what about keeping this functionality baking for a while until when [microsoft/vscode#122071](https://github.com/microsoft/vscode/issues/122071) is fixed?
I mean, you won't get the expected output, and many users may consider it a bug and will start creating support issues, etc.
If just a file is opened, the command `'open in new window'` for a favorite file would produce the same result already.

**Comment by fredjeck** [comment]
The behavior you are describing is odd, I am test driving the exact same version on all my machines and cannot reproduce it (VSCode 1.65.1). May I ask you to uninstall the extension (your favorites will not be removed), restart VScode and then from the extensions panel, use the three dots menu and install the preview version again using the "Install from VSIX" menu ?
Can you give me more details about your environment (OS, type of folder you favorited - local, remote etc..)

**Comment by carloscadux** [comment]
I recorded my screen detailing the issues: https://bit.ly/3tRIEZZ

My environment: 
OS Windows 11 Home 64bit (version 21H2)
Local Folder favorited

If you need further information, just let me know.

**Comment by fredjeck** [comment]
Nailed it !

Thanks a lot for the time you took to create all those screencasts 🥇 , helped me quite a alot and in the end it's not a bug, it's a feature :)

In your tests you are creating extension-less files, however the file filter you are setting is _*.*_ as I am using glob pattern to filter the files. There are no resulting matches (there is no dot character in your file names). I you switch the file filter to _*_ it should work. I will improve the file filter documentation box with some extra examples and change the default filter to * so that it's more obvious.

**Comment by carloscadux** [comment]
> [...]  in the end it's not a bug, it's a feature :)
> [...] I you switch the file filter to _*_ it should work. 

Indeed, it worked 🙃
Handy feature by the way!

**Comment by fredjeck** [comment]
I just released 1.7.0-b02.
I slightly improved the documentation on glob patterns - as behind the scene it is using a real globbin library you can do some pretty fancy things (see https://github.com/isaacs/node-glob)

Also to avoid issues when opening folders locally or in a new window, the extension now warns you when your filter does not matches any files - this can avoid some mistakes :grinning:
